### ISSUE TITLE
Updating Sveltekit/Vite now throws: `request in handle has been replaced with event. See https://github.com/sveltejs/kit/pull/3384 for details`

### ISSUE THREAD

**Issue by inspiredlabs** [FunMiles/SveltekitPWA#9]
I am trying to use SveltekitPWA with the Vite/Svelte inspector, which is available using an entry to `svelte.config.js`:

```
/** @type {import('@sveltejs/kit').Config} */
const config = {
  experimental: { inspector: true },
...
```

However, merging SveltekitPWA with the latest Sveltekit, highlights breaking changes – notably: `hooks.ts` throws: [https://github.com/sveltejs/kit/pull/3384](https://github.com/sveltejs/kit/pull/3384).

Have you had a chance to look at the new interface?

Repo: https://github.com/inspiredlabs/inspector
~~demo: https://offline-phi.vercel.app~~

**Comment by FunMiles** [comment]
@inspiredlabs Yes there are some incompatibilities with the new sveltekit. I have done fixes on a local copy so far but have been a bit too busy to complete a clean work to push back. Hopefully, I'll find time this weekend. I'll try to give more info on here in a few hours if you want to put them in by hand yourself in your project.

**Comment by FunMiles** [comment]
I have a working local fixed version. I need to check it works correctly as a PWA and then I will push it.

**Comment by inspiredlabs** [comment]
> PS: I pushed it but cannot deploy it to Vercel for testing. It builds correctly locally but not on Vercel.

I got the deployment working on [Vercel](https://inspectort.vercel.app/) following a Sveltekit installation and merge – see: `app.html` now has `%sveltekit.things%` as opposed to `%svelte.things%`.

Although now dev tools Lighthouse hangs.



throw new Error("`timestamp` has been removed from $service-worker. Use `version` instead");

**Comment by FunMiles** [comment]
Two things:
- Can you create a pull request with the fixes you list?
- My inability to test it seems to be a known issue. I emailed vercel's support asking how to create a _fresh or clean_ deploytment but have not yet heard from them Your deployment works because it is a fresh deployment, I presume. Mine, trying to update the existing one, fails. See this post in a thread on the subject about it: https://github.com/sveltejs/kit/issues/4135#issuecomment-1059348065

**Comment by inspiredlabs** [comment]
> PS: Third thing, I am not seeing %svelte.things% in the code. Is that something purely of yours? I did change a few svelte into sveltekit, but not with things.

Mea culpa, for the shorthand, what I meant was: `%sveltekit.head%` and `%sveltekit.body%` from this working version of [SveltekitPWA](https://github.com/inspiredlabs/offline/blob/master/src/app.html) which has `%svelte.head%` and `%svelte.body%`, obv. this no longer works as it did. Which you changed.

**Comment by inspiredlabs** [comment]
A PR won't work as there are many changes to Sveltekit.

Here's how to get it working:

- [start a new project](https://kit.svelte.dev/), which will deploy to Vercel OOTB.
- open the contents of your zip: `~/SveltekitPWA-main`, and merge the following:
  - your up-to-date project files (`hooks.js`, `ReloadPrompt/index.svelte`, etc).
  - `svelte.config.js` and comment out `target: '#svelte',`
  - OPTIONAL: I also added: `const config = { experimental: { inspector: true }, ... }`, to `svelte.config.js`, although this doesn't affect the offline capabilities.
  - open `app.html` and ensure `%sveltekit.head%` and `%sveltekit.body%` are there.
  - replace instances of `timestamp` with `version` in [service-worker.ts](https://github.com/inspiredlabs/inspector/blob/master/src/service-worker.ts)
  - update, as per: `[package.json](https://github.com/inspiredlabs/inspector/blob/master/package.json)`.
  - `npm install`
  - you might need to update: `npm install --save core-js@^3` separately. 
  - `npm build && npm run preview`
  - deploy to Vercel with a temporary name, and test it works.
  - delete your old "nice-name" deploy and redeploy with the nice-name.

Feel free to [clone this](https://github.com/inspiredlabs/inspector) if I have missed something in the instructions.

**Comment by FunMiles** [comment]
I have done some changes to the config.json and it seems it does address my deployment issues. I will push the changes after a bit of clean up.

**Comment by FunMiles** [comment]
The changes have been pushed. Unless you see anything else, I will close this issue tomorrow.

**Comment by FunMiles** [comment]
Closing this issue.
